targets/TARGET_Realtek/TARGET_AMEBA/flash_api.c

```diff
@@ -13,555 +13,62 @@
-                // Wait spic busy done
-            SpicWaitBusyDoneRtl8195A();
-            // Wait flash busy done (wip=0)
-            if(flashtype == FLASH_MICRON){
-                SpicWaitOperationDoneRtl8195A(flashobj.SpicInitPara);
-            } else
-                SpicWaitWipDoneRefinedRtl8195A(flashobj.SpicInitPara);
-
-            address += ProgramSize;
-            data+= ProgramSize;   
-            Length -= ProgramSize;
-            OccuSize = 0;
-        } else{
-            while((flashobj.Length) >= PageSize){
-                SpicUserProgramRtl8195A(data, flashobj.SpicInitPara, address, &(flashobj.Length));
-                    // Wait spic busy done
-                SpicWaitBusyDoneRtl8195A();
-                // Wait flash busy done (wip=0)
-                if(flashtype == FLASH_MICRON){
-                    SpicWaitOperationDoneRtl8195A(flashobj.SpicInitPara);
-                } else
-                    SpicWaitWipDoneRefinedRtl8195A(flashobj.SpicInitPara);
-
-                address += PageSize;
-                data+=PageSize;
-                Length -= PageSize;
-            }
-            flashobj.Length = Length;
-            if((flashobj.Length)  > 0){
-                SpicUserProgramRtl8195A(data, flashobj.SpicInitPara, address, &(flashobj.Length));
-                    // Wait spic busy done
-                SpicWaitBusyDoneRtl8195A();
-                // Wait flash busy done (wip=0)
-                if(flashtype == FLASH_MICRON){
-                    SpicWaitOperationDoneRtl8195A(flashobj.SpicInitPara);
-                } else
-                    SpicWaitWipDoneRefinedRtl8195A(flashobj.SpicInitPara);
-
-                break;
-            }
-        }
-        flashobj.Length = Length;
-    }
-    
-    SpicDisableRtl8195A();
-    return 1;
+    return 0;
 }
-/**
-  * @brief  Read a stream of data from specified address
-  * @param  obj: Specifies the parameter of flash object.
-  * @param  address: Specifies the address to be read.
-  * @param  len: Specifies the length of the data to read.
-  * @param  data: Specified the address to save the readback data.
-  * @retval   status: Success:1 or Failure: Others.
-  */
 
-int  flash_burst_read(flash_t *obj, uint32_t address, uint32_t Length, uint8_t * data)
+int32_t flash_program_page(flash_t *obj, uint32_t address, const uint8_t *data, uint32_t size)
 {
-    flash_turnon();
-
-    if(isinit == 0)
-        flash_init(&flashobj);
-
-    // Wait flash busy done (wip=0)
-    SpicWaitWipDoneRefinedRtl8195A(flashobj.SpicInitPara);
-    SpicUserReadRtl8195A(Length, address, data, SpicOneBitMode);
-    SpicDisableRtl8195A();
-    return 1;
+    return __flash_ext_stream_write(obj, FLASH_OFS(address), size, data);
 }
 
-int flash_get_status(flash_t *obj)
+uint32_t flash_get_sector_size(const flash_t *obj, uint32_t address)
 {
-    u8 Status = 0;
+    if (address < FLASH_START || address >= FLASH_END)
+        return (uint32_t)-1;
```

Comment:
Is this value being returned as an error value ? If this is the case, would an explicit sector size of 0 not be better in this case ?